### PR TITLE
Don't calculate cache invalidation if parsed files haven't been read yet

### PR DESCRIPTION
**Pull request by yanaise**
Handled a case where parsed files weren't loaded yet, which caused a cache invalidation which was unnecessary. Users won't really notice this, but it can affect performance

---
There are all sorts of cases in which cache is invalidated even though there's no 
---
_Release Notes_: 
---
_User Notifications_:

### CODE REVIEW

File: packages/workspace/src/workspace/nacl_files/nacl_files_source.ts [hunk 1]
```diff
@@ -269,7 +269,10 @@ const buildNaclFilesState = async ({
   currentState: NaclFilesState
 }): Promise<buildNaclFilesStateResult> => {
   const preChangeHash = await currentState.metadata.get(HASH_KEY)
-  const cacheValid = (preChangeHash === await currentState.parsedNaclFiles.getHash())
```
Comment (roironn) [inline packages/workspace/src/workspace/nacl_files/nacl_files_source.ts:272]: In which flows will this be valid?
As far as I can understand, `getHash` will only return undefined if there are no files... 
So either we have an issue with the hash values which are created here (maybe a serialization issue? Is there a chance we serialize `undefined` as `'undefined'`? ) or we need to better understand why we ended up in this situation IMO

**Review by roironn — COMMENTED** [review]
See inline comment (I started writing this here, but than moved it to the code and now I can't edit :) )

File: packages/workspace/src/workspace/nacl_files/nacl_files_source.ts (hunk 1 above)
Comment (yanaise) [reply packages/workspace/src/workspace/nacl_files/nacl_files_source.ts:272]: @adamro @roironn answering both of you here. Yes, getHash returns undefined if there are no files. This seems to be linked to the following comment on line 397:
// This is temp and should be removed when we change the init flow
// This happens now cause we get here with ParsedNaclFiles that originate from the cache
Followed by this code:
if (values.isDefined(naclFile.buffer)) {
        await currentState.parsedNaclFiles.put(naclFile.filename, naclFile)
      }

Basically when this happens, the parsedNaclFiles are not initialized, and can't be used to calculate hash.

**Comment by netama** [comment]
@yanaise @roironn @adamro is this still relevant or should we close it?
